game.js

```diff
@@ -63,6 +74,93 @@ document.querySelector('.restart').addEventListener('click', () => {
         cell.classList.remove('disabled', 'x', 'o')
     })
 
+    game.xTurn = true
+    game.xState = []
+    game.oState = []
+})
+
+document.querySelector('.return-one-step').addEventListener('click', () => {  
+    const xLength = game.xState.length;
+    const oLength = game.oState.length;
+    const lastXCell = document.querySelector(`[data-value = '${game.xState[xLength -1]}']`)
+    const lastOCell = document.querySelector(`[data-value = '${game.oState[oLength -1]}']`)
+ if (xLength > 0){
+    if (game.xTurn) { 
+     lastOCell.classList.remove('o', 'disabled');
+     game.oState.pop();
+    game.xTurn = false;
+ }
+ else{
+     lastXCell.classList.remove('x', 'disabled');
+     game.xState.pop();
+     game.xTurn = true;
+ }}
+ })
+
+ document.querySelector('.show-record').addEventListener('click', () => {
+ const scoresArr = [];
+
+ for (const score in game.winnings) {
+  scoresArr.push([score, game.winnings[score]]);
+ }
+ if (scoresArr.length > 0){
+scoresArr.sort((a , b) =>  a[1] - b[1])
+alert(`The fastest game was game ${scoresArr[0][0]} with ${scoresArr[0][1]} moves`);
+}
+else {
+    alert("No winnings yet")
+}
+})
+
+document.querySelector('.save-game').addEventListener('click', function(){
+   if(game.xState.length){
+        game.savedGame.states.push(game.xState, game.oState);
+        game.savedGame.xTurn = game.xTurn;
+       this.disabled = true;
```

Comment:
avoid using `this` when it's possible (you can use the event object to detect the element or vice versa).

Reply:
I decided not to go with this.disabled option eventually.
Why avoid using 'this'?

Reply:
`this` can reference to multiple things in javascript ([see here](https://www.javascripttutorial.net/javascript-this/)).
and your code can break if someone will change the code later to arrow function for example.